src/herder/test/HerderTests.cpp

```diff
@@ -3304,6 +3308,52 @@ TEST_CASE("SCP checkpoint", "[catchup][herder]")
         REQUIRE(bufferedLedgers.crbegin()->first ==
                 mainNode->getLedgerManager().getLastClosedLedgerNum());
     }
+
+    SECTION("Two out of sync nodes receive checkpoint")
+    {
+        // Start two out of sync nodes
+        auto outOfSync1 = simulation->addNode(v1SecretKey, qSet, &cfg2);
+        auto outOfSync2 = simulation->addNode(v2SecretKey, qSet, &cfg3);
+
+        simulation->addPendingConnection(v0NodeID, v1NodeID);
+        simulation->addPendingConnection(v0NodeID, v2NodeID);
+
+        simulation->startAllNodes();
+        auto& cm1 =
+            static_cast<CatchupManagerImpl&>(outOfSync1->getCatchupManager());
+        auto& cm2 =
+            static_cast<CatchupManagerImpl&>(outOfSync2->getCatchupManager());
+
+        // Crank until outOfSync node has recieved checkpoint ledger and started
+        // catchup
+        auto f = [&]() {
+            simulation->crankUntil(
+                [&]() {
+                    return cm1.isCatchupInitialized() &&
+                           cm1.getCatchupWorkState() ==
+                               BasicWork::State::WORK_RUNNING &&
+                           cm2.isCatchupInitialized() &&
+                           cm2.getCatchupWorkState() ==
+                               BasicWork::State::WORK_RUNNING;
+                },
+                2 * Herder::SEND_LATEST_CHECKPOINT_DELAY, false);
+        };
+
+        // History archves have not been configured, so this should throw once
+        // catchup starts
+        REQUIRE_THROWS_WITH(f(), "No GET-enabled history archive in config");
+
+        auto const& bufferedLedgers1 = cm1.getBufferedLedgers();
+        REQUIRE(!bufferedLedgers1.empty());
+        REQUIRE(bufferedLedgers1.begin()->first == firstCheckpoint);
```

Comment:
hmm, looks like this condition is failing
```

2	herder/test/HerderTests.cpp:3348: FAILED:
2	
2	  REQUIRE( bufferedLedgers1.begin()->first == firstCheckpoint )
2	
2	with expansion:
2	
2	  85 == 64
2	
```
This suggests that the out of sync node either didn't receive the checkpoint ledger or dropped it by the time we reached this check. Could you verify that both nodes actually receive the checkpoint ledger (IOW, that the main node calls this lambda for both out of sync nodes: https://github.com/stellar/stellar-core/blob/2109a168a895349f87b502ae3d182380b378fa47/src/herder/HerderImpl.cpp#L867)

Reply:
> hmm, looks like this condition is failing
> 
> ```
> 
> 2	herder/test/HerderTests.cpp:3348: FAILED:
> 2	
> 2	  REQUIRE( bufferedLedgers1.begin()->first == firstCheckpoint )
> 2	
> 2	with expansion:
> 2	
> 2	  85 == 64
> 2	
> ```
> 
> This suggests that the out of sync node either didn't receive the checkpoint ledger or dropped it by the time we reached this check. Could you verify that both nodes actually receive the checkpoint ledger (IOW, that the main node calls this lambda for both out of sync nodes:
> 
> https://github.com/stellar/stellar-core/blob/2109a168a895349f87b502ae3d182380b378fa47/src/herder/HerderImpl.cpp#L867
> 
> )

`        auto f = [&]() {
            simulation->crankUntil(
                [&]() {
                    return cm1.isCatchupInitialized() &&
                           cm1.getCatchupWorkState() ==
                               BasicWork::State::WORK_RUNNING &&
                           cm2.isCatchupInitialized() &&
                           cm2.getCatchupWorkState() ==
                               BasicWork::State::WORK_RUNNING;
                },
                2 * Herder::SEND_LATEST_CHECKPOINT_DELAY, false);
        };
`

Is the crankUntil condition correct? I assume it checks that both nodes received catch up messages.

Reply:
> 

I verified that the main node calls the lambda for both out of sync nodes.

Reply:
It looks like the `SCP checkpoint` tests are still failing. I think this check isn't working `cm1.isCatchupInitialized() && cm1.getCatchupWorkState() == BasicWork::State::WORK_RUNNING` now that you've added the history archive config. It looks like the catchup state is stuck in `State::WORK_WAITING` and causing the `crankUntil` loop to timeout. I'm not 100% sure so maybe @marta-lokhova can follow up, but I think `getCatchupWorkState()` goes into `State::WORK_WAITING`while waiting for the history archive download. This didn't happen before because we didn't have a history archive to download from previously. If you change the loop to this it should avoid the timeout:
```
        simulation->crankUntil(
            [&]() {
                return cm.isCatchupInitialized();
            },
            2 * Herder::SEND_LATEST_CHECKPOINT_DELAY, false);
```

Reply:
yeah, @SirTyson is right, removing `getCatchupWorkState` check should fix it (we still ensure that catchup was started with `isCatchupInitialized` check)

Reply:
Thank you @SirTyson  and @marta-lokhova  Test will pass once ```getCatchupWorkState() ==BasicWork::State::WORK_RUNNING``` is deleted. Are we ready to merge this PR after that?

Reply:
I just made a new commit removing the getCatchupWorkState check. Interestingly on my local machine that condition doesn't cause the test to fail?